carma-messenger-core/cpp_message/src/MobilityRequest_Message.cpp

```diff
@@ -0,0 +1,538 @@
+        for(size_t i=0;i<params_string_size;i++)
+        {
+            string_content_params[i]=plainMessage.strategy_params[i];
+        }
+        message->value.choice.TestMessage00.body.strategyParams.buf=string_content_params;
+        message->value.choice.TestMessage00.body.strategyParams.size=params_string_size;
+        
+        //Trajectory
+            //trajectoryStart
+        MobilityLocation* trajectory_location;
+        trajectory_location=(MobilityLocation*)calloc(1,sizeof(MobilityLocation));
+        if(!trajectory_location)
+        {
+            ROS_WARN_STREAM("Cannot allocate mem for trajectory.location encoding");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        long trajectory_start;
+        trajectory_start=plainMessage.trajectory.location.ecef_x;
+        if(trajectory_start> LOCATION_MAX || location_val<LOCATION_MIN){
+            ROS_WARN_STREAM("Location ecefX is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        trajectory_location->ecefX=trajectory_start;
+        
+        trajectory_start=plainMessage.trajectory.location.ecef_y;
+        if(trajectory_start> LOCATION_MAX || trajectory_start<LOCATION_MIN){
+            ROS_WARN_STREAM("Location ecefY is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        trajectory_location->ecefY=trajectory_start;
+
+        trajectory_start=plainMessage.trajectory.location.ecef_z;
+        if(trajectory_start> LOCATION_MAX_Z || trajectory_start<LOCATION_MIN_Z){
+            ROS_WARN_STREAM("Location ecefX is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        trajectory_location->ecefZ=trajectory_start;
+
+        //get location timestamp and convert to char array
+        uint64_t trajectory_time=plainMessage.trajectory.location.timestamp;
+        std::string trajectory_timestamp=std::to_string(trajectory_time);
+        size_t trajectory_timestamp_string_size=trajectory_timestamp.size();
+        uint8_t string_trajectory_timestamp[trajectory_timestamp_string_size];
+        for(size_t i=0;i<trajectory_timestamp_string_size;i++)
+        {
+            string_trajectory_timestamp[i]=trajectory_timestamp[i];
+        }
+        trajectory_location->timestamp.buf=string_trajectory_timestamp;
+        trajectory_location->timestamp.size=trajectory_timestamp_string_size;
+        message->value.choice.TestMessage00.body.trajectoryStart=trajectory_location;
+
+            //trajectory offsets
+        size_t offset_count=plainMessage.trajectory.offsets.size();
+        
+        if(offset_count>MAX_POINTS_IN_MESSAGE){
+            ROS_WARN_STREAM("offset count greater than 60.");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+
+        MobilityLocationOffsets* offsets_list;
+        offsets_list=(MobilityLocationOffsets*)calloc(1,sizeof(MobilityLocationOffsets));
```

Comment:
Same thing as above re: calloc v new. I'd recommend replacing the C-style memory management w/ C++ style news where possible.